### PR TITLE
add more test case for upgrade and br

### LINKED ISSUE #4064: New test cases for TiDB v5.1
## Feature Request

**Is your feature request related to a problem? Please describe:**

- Upgrade cases (Refer to https://github.com/pingcap/tidb-operator/pull/3980)
  - From v4.0.9 to v5.1.0
  - From v5.0.0 to v5.1.0
  - From v5.0.2 to v5.1.0
- BR cases (Refer to https://github.com/pingcap/tidb-operator/pull/3986)
  - Backup with v4.0.9 and restore to v5.1.0
  - Backup with v5.0.0 and restore to v5.1.0
  - Backup with v5.0.2 and restore to v5.1.0
  - Backup with v5.1.0 and restore to v5.1.0

**Describe the feature you'd like:**

**Describe alternatives you've considered:**

**Teachability, Documentation, Adoption, Migration Strategy:**

### PR DESCRIPTION
**Pull request by KanShiori**
### What problem does this PR solve?

Closes  #4064

### Code changes

- [ ] Has Go code change
- [x] Has CI related scripts change

### Tests

- [ ] Unit test 
- [x] E2E test 
- [ ] Manual test 
- [ ] No code 

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [ ] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note

```

### CODE REVIEW

File: tests/e2e/util/image/image.go [hunk 1]
```diff
@@ -27,11 +27,18 @@ import (
...
 
 const (
-	TiDBV3                        = "v3.0.20"
-	TiDBV4                        = "v4.0.12"
-	TiDBV5Prev                    = "v5.0.0"
-	TiDBV5                        = "v5.0.1"
-	TiDBNightlyVersion            = "nightly"
+	// TiDB Version
+	TiDBV3             = "v3.0.20"
+	TiDBV4             = "v4.0.12"
```
Comment (DanielZhangQD) [inline tests/e2e/util/image/image.go:32]:
```suggestion
	TiDBV4             = "v4.0.13"
```

File: tests/e2e/util/image/image.go [hunk 2]
```diff
@@ -27,11 +27,18 @@ import (
...
+	TiDBV5Prev         = "v5.0.0"
+	TiDBV5             = "v5.0.1"
```
Comment (DanielZhangQD) [inline tests/e2e/util/image/image.go:34]:
```suggestion
	TiDBV5             = "v5.0.3"
```

File: tests/e2e/util/image/image.go [hunk 3]
```diff
@@ -27,11 +27,18 @@ import (
...
+	TiDBNightlyVersion = "nightly"
+	// specific version
+	TiDBV4x0x9 = "v4.0.9"
+	TiDBV5x0x0 = "v5.0.0"
+	TiDBV5x0x2 = "v5.0.2"
+	TiDBV5x1x0 = "v5.1.0"
```
Comment (DanielZhangQD) [inline tests/e2e/util/image/image.go:40]:
```suggestion
	TiDBV40InUse0 = "v4.0.9"
	TiDBV50InUse0 = "v5.0.0"
	TiDBV50InUse1 = "v5.0.2"
	TiDBV5xLatestStable = "v5.1.0"
```

Comment (DanielZhangQD) [reply tests/e2e/util/image/image.go:40]: To avoid updating the cases when the version updates.

**Review by DanielZhangQD — APPROVED** [review]
LGTM

**Review by csuzhangxc — APPROVED** [review]

**Pull request merged**